figures/mapping-histology-labels.Rmd

```diff
@@ -0,0 +1,240 @@
+- `display_group` - the high-level histology labels that should be used for plotting   
+- `hex_codes` the direct colors that should be used for plotting  
+
+With this info, `histology_label_color_table.tsv` can be used by all plots and figures that summarize high level data  while displaying histology information. 
+
+# How `display_group` is made:
+
+Here's how `broad-histology` groups are [combined into the higher-level groupings of `display_group`](#declare-new-equivalent-groups).
+
+1) "Lymphoma", "Melanocytic tumor", "Other tumor", "Metastatic tumors", "Non-CNS tumor" are combined into a `Other non-CNS group` in `display_group`. 
+
+2) `Benign tumor` and `Non-tumor` biospecimens are combined into a `Benign` group. 
+
+3) `Other astrocytic tumor` samples are combined into the existing `Low-grade astrocytic tumor`.
+
+4) Anything not in the above categories gets its `broad_histology` label carried over. 
+
+# Usage
+
+This notebook can be run via the command line from the top directory of the 
+repository as follows:
+
+```
+Rscript -e "rmarkdown::render('figures/mapping-histology-labels.Rmd', 
+                              clean = TRUE)"
+```
+
+## Set Up
+
+```{r}
+# Magrittr pipe
+`%>%` <- dplyr::`%>%`
+```
+
+### Directories and Files
+
+```{r}
+# Path to input directory
+input_dir <- file.path("..", "data")
+output_dir <- "palettes"
+```
+
+# Read in metadata 
+
+Which variables are we keeping for this table? 
+
+```{r}
+histology_variables <- 
+  c("integrated_diagnosis", 
+    "Notes", 
+    "harmonized_diagnosis",
+    "broad_histology", 
+    "short_histology")
+```
+
+Let's read in the current release's `pbta-histologies.tsv` file. 
+
+```{r}
+metadata <-
+  readr::read_tsv(file.path(input_dir, "pbta-histologies.tsv"), guess_max = 10000)
+```
+
+Now we'll select histology variables we mentioned above and so capitalization differences don't get in the way with this process, we will change everything to lower case for now. 
+
+```{r}
+working_metadata <- metadata %>% 
+  dplyr::select(Kids_First_Biospecimen_ID, sample_type, histology_variables) %>% 
+  dplyr::mutate_at(histology_variables, tolower)
+```
+
+# Take a look at how many biospecimens per `broad_histology` group
+
+Let's summarize `broad_histology`. 
+Because the `Normal` samples don't have histologies, we'll look at just the `Tumor` samples at for this summary. 
+
+```{r}
+broad_summary <- working_metadata %>% 
+  dplyr::filter(sample_type == "Tumor") %>%
+  dplyr::count(broad_histology) %>% 
+  dplyr::arrange(n)
+```
+
+Let's print out the summary. 
+
+```{r}
+broad_summary %>% 
+  knitr::kable()
+```
+
+There's handful of very small groups (many are n = 2). 
+
+## Declare new equivalent groups
+
+These groups we'll combine into a non-CNS/other tumor group.
+
+```{r}
+other_tumor <- c("lymphoma", "melanocytic tumor", "other tumor", "metastatic tumors", "non-cns tumor")
+```
+
+These groups we'll combine as a benign.
+
+```{r}
+begnign <- c("benign tumor", "non-tumor")
```

Comment:
Typo "benign"

Reply:
My brain kept looking at this when I wrote it because I thought it was wrong but then my brain convinced me it was right. LOL